channel.go

```diff
@@ -57,7 +57,9 @@ type ChannelOptions struct {
 	// The name of the process, for logging and reporting to peers
 	ProcessName string
 
-	// OnPeerStatusChanged
+	// OnPeerStatusChanged is an optional callback that receives a notification
```

Comment:
Want to discuss whether we want to report other states:
 - peer that received a new connection (especially if it's from 0 -> 1)
 - peer that went from N -> N-1 where N > 2.

I think we want the former, but not the latter.